### PR TITLE
Vaibhav/POS 325

### PR DESCRIPTION
**Pull request by VAIBHAVJINDAL3012**
https://linear.app/matic/issue/POS-325/improve-logging-in-heimdall-bridge-cli-to-better-debug-issues 
Improved the error logging in the Heimdall

### CODE REVIEW

**Comment by VAIBHAVJINDAL3012** [comment]
Please review the request @temaniarpit27

**Comment by temaniarpit27** [comment]
@VAIBHAVJINDAL3012 resolve conflicts

File: bridge/setu/processor/checkpoint.go [hunk 1]
```diff
@@ -635,7 +635,7 @@ func (cp *CheckpointProcessor) proposeCheckpointNoAck() (err error) {
 
 	// return broadcast to heimdall
 	if err := cp.txBroadcaster.BroadcastToHeimdall(msg); err != nil {
-		cp.Logger.Error("Error while broadcasting checkpoint-no-ack to heimdall", "error", err)
+		cp.Logger.Error("Error while broadcasting checkpoint-no-ack to heimdall", "error", err, "msg", msg)
```
Comment (temaniarpit27) [inline bridge/setu/processor/checkpoint.go:638]: check if msg is getting correctly printed or not

**Review by temaniarpit27 — DISMISSED** [review]

File: bridge/setu/listener/heimdall.go [hunk 2]
```diff
@@ -75,7 +75,7 @@ func (hl *HeimdallListener) StartPolling(ctx context.Context, pollInterval time.
 		case <-ticker.C:
 			fromBlock, toBlock, err := hl.fetchFromAndToBlock()
 			if err != nil {
-				hl.Logger.Error("Error fetching fromBlock and toBlock...skipping events query", "error", err)
+				hl.Logger.Error("Error fetching fromBlock(", fromBlock, ") and toBlock(", toBlock, ")...skipping events query", "error", err)
```
Comment (manav2401) [inline bridge/setu/listener/heimdall.go:78]: Do something like
`hl.Logger.Error("Error fetching from and to block, skipping events query", "fromBlock", fromBlock, "toBlock", toBlock, "error", err)`

File: checkpoint/genesis.go [hunk 3]
```diff
@@ -30,7 +30,10 @@ func InitGenesis(ctx sdk.Context, keeper Keeper, data types.GenesisState) {
 		for i, checkpoint := range data.Checkpoints {
 			checkpointIndex := uint64(i) + 1
 			if err := keeper.AddCheckpoint(ctx, checkpointIndex, checkpoint); err != nil {
-				keeper.Logger(ctx).Error("InitGenesis | AddCheckpoint", "error", err)
+				keeper.Logger(ctx).Error("InitGenesis | AddCheckpoint",
+					"error", err,
+					"checkpointIndex", checkpointIndex,
+					"checkpoint", checkpoint)
```
Comment (manav2401) [inline checkpoint/genesis.go:36]: Use checkpoint.String()

File: checkpoint/handler.go [hunk 4]
```diff
@@ -44,7 +44,7 @@ func handleMsgCheckpointAdjust(ctx sdk.Context, msg types.MsgCheckpointAdjust, k
 
 	checkpointObj, err := k.GetCheckpointByNumber(ctx, msg.HeaderIndex)
 	if err != nil {
-		logger.Error("Unable to get checkpoint from db", "error", err)
+		logger.Error("Unable to get checkpoint from db", "error", err, "Header Index-", msg.HeaderIndex)
```
Comment (manav2401) [inline checkpoint/handler.go:47]: Refactor this to:
`logger.Error("Unable to get checkpoint from db", "header index", msg.HeaderIndex, "error", err)`

File: checkpoint/handler.go [hunk 5]
```diff
@@ -118,7 +118,7 @@ func handleMsgCheckpoint(ctx sdk.Context, msg types.MsgCheckpoint, k Keeper, con
 			return common.ErrDisCountinuousCheckpoint(k.Codespace()).Result()
 		}
 	} else if err.Error() == common.ErrNoCheckpointFound(k.Codespace()).Error() && msg.StartBlock != 0 {
-		logger.Error("First checkpoint to start from block 0", "Error", err)
+		logger.Error("First checkpoint to start from block 0", "Error", err, "Checkpoint start block", msg.StartBlock)
```
Comment (manav2401) [inline checkpoint/handler.go:121]: Make error as the last parameter and use lowercase for keys

File: checkpoint/handler.go [hunk 6]
```diff
@@ -246,7 +246,9 @@ func handleMsgCheckpointNoAck(ctx sdk.Context, msg types.MsgCheckpointNoAck, k K
 
 	// If last checkpoint is not present or last checkpoint happens before checkpoint buffer time -- thrown an error
 	if lastCheckpointTime.After(currentTime) || (currentTime.Sub(lastCheckpointTime) < bufferTime) {
-		logger.Debug("Invalid No ACK -- Waiting for last checkpoint ACK")
+		logger.Debug("Invalid No ACK -- Waiting for last checkpoint ACK", "lastCheckpointTime:", lastCheckpointTime, "Current Time", currentTime,
```
Comment (manav2401) [inline checkpoint/handler.go:249]: Remove colons and use lowercase.

File: checkpoint/handler.go [hunk 7]
```diff
@@ -255,7 +257,8 @@ func handleMsgCheckpointNoAck(ctx sdk.Context, msg types.MsgCheckpointNoAck, k K
 	lastNoAckTime := time.Unix(int64(lastNoAck), 0)
 
 	if lastNoAckTime.After(currentTime) || (currentTime.Sub(lastNoAckTime) < bufferTime) {
-		logger.Debug("Too many no-ack")
+		logger.Debug("Too many no-ack", "lastNoAckTime", lastNoAckTime, "Current Time", currentTime,
```
Comment (manav2401) [inline checkpoint/handler.go:260]: Similar modifications as above

File: checkpoint/side_handler.go [hunk 8]
```diff
@@ -48,13 +48,13 @@ func SideHandleMsgCheckpointAdjust(ctx sdk.Context, k Keeper, msg types.MsgCheck
 
 	checkpointObj, err := k.GetCheckpointByNumber(ctx, msg.HeaderIndex)
 	if err != nil {
-		logger.Error("Unable to get checkpoint from db", "error", err)
+		logger.Error("Unable to get checkpoint from db", "error", err, "Header Index:", msg.HeaderIndex)
```
Comment (manav2401) [inline checkpoint/side_handler.go:51]: Similar modifications as above

File: checkpoint/side_handler.go [hunk 9]
```diff
@@ -48,13 +48,13 @@ func SideHandleMsgCheckpointAdjust(ctx sdk.Context, k Keeper, msg types.MsgCheck
...
 		return common.ErrorSideTx(k.Codespace(), common.CodeNoCheckpoint)
 	}
 
 	rootChainInstance, err := contractCaller.GetRootChainInstance(chainParams.RootChainAddress.EthAddress())
 	if err != nil {
-		logger.Error("Unable to fetch rootchain contract instance", "error", err)
+		logger.Error("Unable to fetch rootchain contract instance", "error", err, "Eth Address", chainParams.RootChainAddress.EthAddress())
```
Comment (manav2401) [inline checkpoint/side_handler.go:57]: Similar modifications as above

File: checkpoint/side_handler.go [hunk 10]
```diff
@@ -124,7 +134,10 @@ func SideHandleMsgCheckpointAck(ctx sdk.Context, k Keeper, msg types.MsgCheckpoi
 
 	rootChainInstance, err := contractCaller.GetRootChainInstance(chainParams.RootChainAddress.EthAddress())
 	if err != nil {
-		logger.Error("Unable to fetch rootchain contract instance", "error", err)
+		logger.Error("Unable to fetch rootchain contract instance",
+			"error", err,
+			"Eth Address", chainParams.RootChainAddress.EthAddress(),
```
Comment (manav2401) [inline checkpoint/side_handler.go:139]: Similar modifications as above

File: checkpoint/side_handler.go [hunk 11]
```diff
@@ -190,7 +214,9 @@ func PostHandleMsgCheckpointAdjust(ctx sdk.Context, k Keeper, msg types.MsgCheck
 
 	checkpointObj, err := k.GetCheckpointByNumber(ctx, msg.HeaderIndex)
 	if err != nil {
-		logger.Error("Unable to get checkpoint from db", "error", err)
+		logger.Error("Unable to get checkpoint from db",
+			"error", err,
+			"Checkpoint Number", msg.HeaderIndex)
```
Comment (manav2401) [inline checkpoint/side_handler.go:219]: Similar modifications as above

**Review by manav2401 — DISMISSED** [review]

**Review by temaniarpit27 — APPROVED** [review]

**Pull request merged**